build/charts/antrea/README.md

```diff
@@ -44,6 +44,7 @@ Kubernetes: `>= 1.16.0-0`
 | antreaProxy.nodePortAddresses | list | `[]` | String array of values which specifies the host IPv4/IPv6 addresses for NodePort. By default, all host addresses are used. |
 | antreaProxy.proxyAll | bool | `false` | Proxy all Service traffic, for all Service types, regardless of where it comes from. |
 | antreaProxy.proxyLoadBalancerIPs | bool | `true` | When set to false, AntreaProxy no longer load-balances traffic destined to the External IPs of LoadBalancer Services. |
+| antreaProxy.serviceProxyName | string | `""` | The value of service.kubernetes.io/service-proxy-name label for AntreaProxy to match. If it is set, then AntreaProxy only handles the Service objects matching this label. The default value is empty string, which means that AntreaProxy will manage all Service objects without the mentioned label. |
```

Comment:
Why we care about the value? Should we just ignore Services with the label set, no matter what is the label value?

Reply:
If serviceProxyName is empty, we should ignore Services with the label set, no matter what is the label value; if serviceProxyName is set like "antrea", then we will handle the Services with the label `service.kubernetes.io/service-proxy-name=antrea`.

Reply:
Is that (filtering only a particular proxy-name) what suggested by the K8s upstream, or implemented by kube-proxy?

Reply:
It was implemented by kube-proxy, see https://github.com/kubernetes/kubernetes/blob/bc41f37e85dd67ddb69eaa9e4da530c3c72101d7/cmd/kube-proxy/app/server.go#L696, and a sig-network e2e test is used to test this feature, see https://github.com/kubernetes/kubernetes/blob/bc41f37e85dd67ddb69eaa9e4da530c3c72101d7/test/e2e/network/service.go#L2230.

Reply:
In the kube-proxy case, it does not care about the label value, but just ignores the Services with the label, no matter what is the label value?

Should we follow the same behavior, as AntreaProxy should be the default proxy when enabled? In which case, you see users may want Antrea to handle only Services with a particular label value like "antrea" (and in that case, what will handle other Services)?

Reply:
> If we set the fault value to 'antrea', then only the Services with the label service.kubernetes.io/service-proxy-name=antrea will be handled by AntreaProxy. Services with the label service.kubernetes.io/service-proxy-name having any other value or without the label will not be handled by AntreaProxy.

I got that. I meant what is the real use case in your mind. Why users want AntreaProxy to handle only Services with a particular label? What proxy will handle other Services (and we do not support another proxy right? Do we?)?

Reply:
When deploying Antrea with other CNIs:

If the `antreaProxy.serviceProxyName` field is left empty, AntreaProxy will handle all Services without a label. To exclude certain Services from being handled by AntreaProxy, users can label those Services with the key `service.kubernetes.io/service-proxy-name` and assign any value to it.

If the `antreaProxy.serviceProxyName` field is set to "antrea", users can selectively assign AntreaProxy to handle only specific Services by labeling them with `service.kubernetes.io/service-proxy-name=antrea`. Other Services should be managed by the other CNIs deployed with Antrea.

Reply:
Ok, you meant we deploy Antrea in NetworkPolicy only mode, with Cilium or Calico, and users want Cilium/Calico to handle some Services? But does that even work?

Reply:
I think it is not very necessary to support matching a particular value, as we should not have a case with another proxy to the default one. But anyway as the code is not complex, I am ok to support it too.